src/evict/evict_lru.c

```diff
@@ -64,13 +64,14 @@ __evict_entry_priority(WT_SESSION_IMPL *session, WT_REF *ref)
 {
     WT_BTREE *btree;
     WT_PAGE *page;
-    uint64_t read_gen;
+    uint64_t page_read_gen, read_gen;
 
     btree = S2BT(session);
     page = ref->page;
 
+    page_read_gen = __wt_atomic_load64(&page->read_gen);
```

Comment:
This is an example where we could upgrade to a `READ_ONCE`. The `WT_READGEN_EVICT_SOON` macro looks like
```
#define WT_READGEN_EVICT_SOON(readgen) \
    ((readgen) != WT_READGEN_NOTSET && (readgen) < WT_READGEN_START_VALUE)
```
and clearly want both occurrences of `readgen` to be the same value.

Similarly I've left the third read of `page->read_gen` below as it's own atomic read. It could probably use the local variable, but I want to avoid behavioural changes in this PR.

Reply:
Yeah I'd put a WT_READ_ONCE in here.

Reply:
Would it make sense to bury the load of page->read_gen into the macro?

Reply:
Yep, `WT_READGEN_EVICT_SOON` has now become `__wt_readgen_evict_soon` in [6a18404](https://github.com/wiredtiger/wiredtiger/pull/10328/commits/6a18404e99790a24e809af814dfcf4fed89d4be2) 👍